_posts/2019-11-19-demo-fraud-detection.md

```diff
@@ -0,0 +1,223 @@
+---
+layout: post
+title: "Building a Fraud Detection System with Runtime Rules Updates on Apache Flink"
+date: 2019-11-19T12:00:00.000Z
+authors:
+- alex:
+  name: "Alexander Fedulov"
+  twitter: "afedulov"
+excerpt:
+categories: news
+---
+
+In this series of blog posts you will learn about three powerful Flink patterns for building streaming applications:
+
+ - Dynamic updates of application logic
+ - Dynamic data partitioning (shuffle), controlled at runtime
+ - Low latency alerting based on custom window logic (not using the Window API)
+
+These patterns expand the possibilities of what is achievable with statically defined data flows and thus provide important building blocks to fulfil versatile business requirements.
+
+**Dynamic updates of application logic** allow Flink jobs to change at runtime, without downtime from stopping and resubmitting the code.  
+<br>
+**Dynamic data partitioning** enables the ability to change how events are being distributed and grouped by the `keyBy()` operator in Flink at runtime. Such functionality often becomes a natural requirement when building jobs with dynamically reconfigurable application logic.  
+<br>
+**Custom window management** demonstrates how you can utilize the low level [ProcessFunction API](https://ci.apache.org/projects/flink/flink-docs-stable/dev/stream/operators/process_function.html), when [Window API](https://ci.apache.org/projects/flink/flink-docs-stable/dev/stream/operators/windows.html) is not exactly matching your requirements. Specifically, you will learn how to implement low latency alerting on windows and how to limit state growth with timers.    
+
+The patterns are built up on top of core Flink functionality, however they might not be immediately obvious from the framework's documentation as it is not trivial to explain and give motivation behind them without a concrete example use case. For this reason we will look at the details of the patterns on an example of building an application which represents a common usage scenario for Apache Flink - a _Fraud Detection_ engine.
+We hope this series will put these powerful approaches into your tool belt and enable you to tackle new tasks with Apache Flink.
+
+In the first blog post of the series we will look at the high level architecture of the demo application, describe its components and their interactions. We will proceed with a deep dive into the implementation details of the first pattern in the series - **dynamic data partitioning**.
+
+You will be able to run the full Fraud Detection Demo application locally and look into the details of the implementation by using the accompanying GitHub repository.
+
+### Fraud Detection Demo
```

Comment:
I would consider changing the subtitle for better SEO to something like: "Apache Flink Demo: Building a Fraud Detection Engine with Flink"

Reply:
Agreed

Reply:
I'm wondering if this section should go first? Introduce the project and then talk about how the reader will learn certain tools when going through the implmentation.

Reply:
Seems contrived as the subtitle.